testing/testcafe/tests/dataGrid/editing.ts

```diff
@@ -2125,24 +2126,64 @@ test('Cells should be focused correctly on click when cell editing mode is used
 });
 
 // T1130497
-test('The first cell of the last row should be focused when newRowPosition = last and editing.mode = cell', async (t) => {
-  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
-  const dataGrid = new DataGrid('#container');
-  const headerPanel = dataGrid.getHeaderPanel();
-
-  await t
-    .click(headerPanel.getAddRowButton())
-    .expect(dataGrid.getDataRow(3).isInserted).ok('row is inserted')
-    .expect(await takeScreenshot('grid-cell-edit-mode-and-new-row-position-last.png', dataGrid.element))
-    .ok()
-    .expect(compareResults.isValid())
-    .ok(compareResults.errorMessages());
-}).before(async () => createWidget('dxDataGrid', {
-  dataSource: [{ name: 'AaAaA', value: 1 }, { name: 'aAaAa', value: 2 }, { name: 'BbBb', value: 3 }],
-  editing: {
-    mode: 'cell',
-    allowUpdating: true,
-    allowAdding: true,
+[
+  {
+    newRowPosition: 'first',
+    insertedRowNumber: 0,
+    scrollMode: 'standard',
+  }, {
     newRowPosition: 'last',
+    insertedRowNumber: 21,
+    scrollMode: 'standard',
+  }, {
+    newRowPosition: 'pageBottom',
+    insertedRowNumber: 21,
+    scrollMode: 'standard',
+  }, {
+    newRowPosition: 'pageTop',
+    insertedRowNumber: 0,
+    scrollMode: 'standard',
+  }, {
+    newRowPosition: 'viewportBottom',
+    insertedRowNumber: 8,
+    scrollMode: 'standard',
+  }, {
+    newRowPosition: 'viewportTop',
+    insertedRowNumber: 0,
+    scrollMode: 'standard',
+  }, {
+    newRowPosition: 'viewportBottom',
+    insertedRowNumber: 8,
+    scrollMode: 'virtual',
+  }, {
+    newRowPosition: 'viewportTop',
+    insertedRowNumber: 0,
+    scrollMode: 'virtual',
   },
-}));
+].forEach((testCase) => {
+  test(`The first cell of the last row should be focused when newRowPosition = ${testCase.newRowPosition} and editing.mode = cell and ${testCase.scrollMode} scroll mode`, async (t) => {
+    const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+    const dataGrid = new DataGrid('#container');
+    const headerPanel = dataGrid.getHeaderPanel();
+
+    await t
+      .click(headerPanel.getAddRowButton())
+      .expect(dataGrid.getDataRow(testCase.insertedRowNumber).isInserted).ok('row is inserted')
+      .expect(await takeScreenshot(`grid-cell-edit-mode-and-new-row-position-${testCase.newRowPosition}-and-${testCase.scrollMode}-scroll-mode.png`, dataGrid.element))
```

Comment:
Minor comment.

Let's change the screenshot file name:

```suggestion
      .expect(await takeScreenshot(`grid-cell-edit-mode-and-new-row_position-${testCase.newRowPosition}_scroll-mode-${testCase.scrollMode}.png`, dataGrid.element))
```

to make it more readable :)